Payment processing and CI improvements

olf42 opened the pull request:
My Draft for payment processing. 

Unfortunately I was not attentive enough, to pack the ci improvements into a separate branch. :/

olf42 committed:
added peewee to requirements

olf42 committed:
fix Makefile docs command

olf42 committed:
update requirements and add pytest-datadir

olf42 committed:
add import-bank-statement command

This cli command imports a bank-statement
into a yaml file, where annotations can be
made, i.e. bookings can be accounted to
customers.

Currently only Postbank is supported.

olf42 committed:
fix gh actions

* flake now really exits non-zero if errors are found
* black exists non-zero if errors are found
* install requirements-dev.txt instead of user requirements

olf42 committed:
flake8 docs/conf.py

olf42 committed:
flake8 files

olf42 committed:
change black-check to style-check

also the flake8 check as it is run
on the ci is added to this command

olf42 committed:
refactor get_settings_from_file

olf42 committed:
black docs/conf.py

olf42 committed:
also run black on the docs dir

olf42 committed:
draft of the report command

*TODO: save reports
*TODO: batch report
*TODO: pdf reports
*TODO: send reports

olf42 committed:
flake and black payment and cli

aparcar commented on .github/workflows/rechnung_checks_and_tests.yml:
```diff
@@ -16,19 +16,18 @@ jobs:
         pip install black
-        black .
+        black --check .
     - name: Test with pytest
```
Instead of having these checks both in the ci and makefile, put it only in the makefile and call the makefile via ci

olf42 replied:
Fixed.

olf42 committed:
don't use pytest-datadir

I suspect this, or the fact that the datadir is
a symlink prevents the ci checks from passing

olf42 committed:
fix gh action to run make style-check

olf42 committed:
add flake8 as dev requirement

olf42 committed:
add print for debugging to failing import test

olf42 committed:
add circleci

aparcar reviewed (commented):
Why adding another CI system?

aparcar commented on rechnung/settings.py:
```diff
@@ -45,6 +45,9 @@
     "policy_attachment_asset_file": "policy.pdf",
+    "bank_statements_dir": "bank_statements",
+    "payments_dir": "payments",
+    "decimal_quantization": ".01",
```
Isn't his obvious and don't need it's own option?

olf42 replied:
But it should be created on init and also should be configurable. I.e. if your bank_statements are stored somewhere else in the system and not necessarily in the applications data folder.

aparcar commented on rechnung/tests/fixtures/payments/postbank.yaml:
```diff
@@ -0,0 +1,32 @@
+- amount: !!python/object/apply:decimal.Decimal
```
I'm again storing Python objects in Yaml as we can't use the yaml safe load mode anymore...

olf42 replied:
Yeah. That's a good point.

aparcar commented on rechnung/transactions.py:
```diff
@@ -1,14 +1,7 @@
-import datetime
```
This whole module should be removed in favor of your new bank statement module.

olf42 replied:
agreed. But as I lack test data, I would prefer that you do it, or submit test data. :)

aparcar commented:
Overall mixing package updates, CI updates and new CSV features in a single request makes it a bit hart to review :)

olf42 commented:
Sorry :(

> Overall mixing package updates, CI updates and new CSV features in a single request makes it a bit hart to review :)

Yeah. I fucked this one up.

olf42 committed:
Merge branch 'develop' into feature/payment_processing
